core/sql/generator/GenPreCode.cpp

```diff
@@ -11274,6 +11274,485 @@ short HbaseAccess::extractHbaseFilterPreds(Generator * generator,
+	for (ValueId vid = preds.init();
+       (preds.next(vid));
+       preds.advance(vid))
+    {
+      ItemExpr * ie = vid.getItemExpr();
+
+      // if it is AND operation, recurse through left and right children
+      if (ie->getOperatorType() == ITM_AND)
+        {
+          ValueIdSet leftPreds;
+          ValueIdSet rightPreds;
+          leftPreds += ie->child(0)->castToItemExpr()->getValueId();
+          rightPreds += ie->child(1)->castToItemExpr()->getValueId();
+          if (isFirstAndLayer){
+            NABoolean leftOK  = extractHbaseFilterPredsV2(generator, leftPreds, newExePreds,TRUE, TRUE);
+            NABoolean rightOK = extractHbaseFilterPredsV2(generator, rightPreds, newExePreds,TRUE, TRUE);
+            if (leftOK && rightOK){
+            	if (!checkOnly){
+					extractHbaseFilterPredsV2(generator, leftPreds, newExePreds,FALSE, TRUE);//generate tree
+					extractHbaseFilterPredsV2(generator, rightPreds, newExePreds,FALSE, TRUE);//generate tree
+					opList_.insert("AND"); // insert an AND node since both side are OK to push down
+					if (addedNode>0)opList_.insert("AND"); // if it is not the first node addd to the push down, AND it with the rest
+					addedNode++; // we just pushed it down, so increase the node count pushed down.
+            	}
+            	if (preds.entries()==1)
+            		return TRUE;
+            } else if (leftOK){  // if only left is OK to push down
+            	if(!checkOnly){
+					extractHbaseFilterPredsV2(generator, leftPreds, newExePreds,FALSE, TRUE);//generate left tree
+					newExePreds.insert(rightPreds); //make sure we add the right child to predicates that needs executor evalvaluation
+					if (addedNode>0)opList_.insert("AND"); // if it is not the first node addd to the push down, AND it with the rest
+					addedNode++; // we pushed down left side so mark it
+            	}
+            	if (preds.entries()==1)
+            		return TRUE;
+            } else if (rightOK){// if only right is OK to push down
+            	if(!checkOnly){
+					extractHbaseFilterPredsV2(generator, rightPreds, newExePreds,FALSE, TRUE);//generate right tree
+					newExePreds.insert(leftPreds);//make sure we add the left child to predicates that needs executor evalvaluation
+					if (addedNode>0)opList_.insert("AND"); // if it is not the first node addd to the push down, AND it with the rest
+					addedNode++;// we pushed down right side so mark it
+            	}
+            	if (preds.entries()==1)
+            		return TRUE;
+            } else{
+            	if(!checkOnly){
+            		newExePreds.insert(vid);// we pushed down nothing, make sure the whole node is evaluated by Executor
+            	}
+            	if (preds.entries()==1)
+            		return FALSE;
+
+            }
+          }
+          else{//if not first AND layer, both left and right must be pushable to get anything pushed
+              if(extractHbaseFilterPredsV2(generator, leftPreds, newExePreds, TRUE, FALSE)&&
```

Comment:
cool, I am thinking adding a CQD to set the MAX OR size hbase push down and default it to conservative 10 until we have more test data to know when it is good? Thought? In a new version we could think about pushing down MIN MAX, and keep the or list at executor level, putting back the column in question in the result set so it can be re evaluated by executor.